### PR TITLE
[KOGITO-4210] Add health extension to Quarkus examples

### PR DESCRIPTION
**Pull request by Kevin-Mok**
See: https://issues.redhat.com/browse/KOGITO-4210

This PR makes the Quarkus examples pass the HTTP liveness/readiness checks when deployed on OpenShift.

## Implementation
I wrote a simple `fish` script to automate adding the extension to the Quarkus directories:
```fish
for dir in (ls -d */)
    # don't match Spring Boot dirs
    if test -z (string match '*springboot*' $dir)
        cd $dir
        and ../mvnw quarkus:add-extension -Dextensions="smallrye-health"

        cd ..
    end
end
```

This removes all comments from the `pom.xml`, so I manually added back the necessary ones.

## Related PR's
- [kogito-cloud-operator](https://github.com/kiegroup/kogito-cloud-operator/pull/733): [KOGITO-4210] Fix probe config for Quarkus deployments
- [kie-docs](https://github.com/kiegroup/kie-docs/pull/3227): [KOGITO-4210] Quarkus apps must have health extension

## Requirements
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by sutaakar** [comment]
@Kevin-Mok The script seems to add
```
      <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-bom</artifactId>
        <version>1.11.0.Final</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
```
IMHO it is not needed as the quarkus bom is transitively imported by kogito-quarkus-bom

**Comment by Kevin-Mok** [comment]
I removed it using another script. :sweat_smile: 
```fish
for pom in (rg -lt xml kogito-quarkus-bom)
    perl -p0e 's/.*<dependency>\n.*\n.*<artifactId>quarkus-bom<\/artifactId>\n.*\n.*\n.*\n.*<\/dependency>\n//g' -i $pom
end
```

File: serverless-workflow-github-showcase/pom.xml [hunk 1]
```diff
@@ -36,12 +36,19 @@
         <artifactId>cloudevents-core</artifactId>
         <version>${version.cloudevents}</version>
       </dependency>
+      <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-universe-bom</artifactId>
+        <version>1.11.0.Final</version>
+        <type>pom</type>
+        <scope>import</scope>
```
Comment (ricardozanini) [inline serverless-workflow-github-showcase/pom.xml]: I think we don't need this. Your script missed this one. :smile:

File: serverless-workflow-temperature-conversion/multiplication-service/pom.xml [hunk 2]
```diff
@@ -1,13 +1,12 @@
 <?xml version="1.0"?>
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (ricardozanini) [inline serverless-workflow-temperature-conversion/multiplication-service/pom.xml]: Actually, we need to be `https` instead. It's a security requirement.

**Review by ricardozanini — COMMENTED** [review]
@Kevin-Mok seems fine. But I'm curious about the dependency management on this huge project. I believe we can add the dependency to the parent `pom`, having it propagated to all modules. This way, people writing new examples won't miss this one. But since it's only for Quarkus, I think we might need to break the example BOM for each runtime. :thinking: 

@danielezonca wdyt?

File: serverless-workflow-github-showcase/pom.xml (hunk 1 above)
Comment (Kevin-Mok) [reply serverless-workflow-github-showcase/pom.xml]: The two containing this ([serverless-workflow-github-showcase](https://github.com/kiegroup/kogito-examples/blob/master/serverless-workflow-github-showcase/pom.xml), [serverless-workflow-temperature-conversion](https://github.com/kiegroup/kogito-examples/blob/master/serverless-workflow-temperature-conversion/pom.xml)) don't have a `kogito-quarkus` or `kogito-quarkus-bom` dependency though. :thinking:

File: kogito-travel-agency/extended/pom.xml [hunk 3]
```diff
@@ -24,11 +25,10 @@
...
-  <modules>
-      <module>travels</module>
-      <module>visas</module>
-  </modules>
-
-
+  <dependencies>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-health</artifactId>
```
Comment (sutaakar) [inline kogito-travel-agency/extended/pom.xml]: This declaration can be removed as the dependency is defined in child modules.

File: kogito-travel-agency/pom.xml [hunk 4]
```diff
@@ -25,11 +25,10 @@
...
-  <modules>
-      <module>basic</module>
-      <module>extended</module>
-  </modules>
-
-
+  <dependencies>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-health</artifactId>
```
Comment (sutaakar) [inline kogito-travel-agency/pom.xml:31]: Same here.

File: onboarding-example/pom.xml [hunk 5]
```diff
@@ -24,7 +27,12 @@
       </dependency>
     </dependencies>
   </dependencyManagement>
-
+  <dependencies>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-health</artifactId>
```
Comment (sutaakar) [inline onboarding-example/pom.xml]: And here.

File: serverless-workflow-github-showcase/pom.xml (hunk 1 above)
Comment (ricardozanini) [reply serverless-workflow-github-showcase/pom.xml]: 'cause they are parent modules, and the sub-modules are not Kogito projects. :sweat:

File: kogito-travel-agency/extended/pom.xml (hunk 3 above)
Comment (ricardozanini) [reply kogito-travel-agency/extended/pom.xml]: Hey @sutaakar! What about keeping in the parents and removing from child modules? The same for the Serverless Workflow examples that have sub-modules.

Comment (sutaakar) [reply kogito-travel-agency/extended/pom.xml]: fine for me

**Review by ricardozanini — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

**Comment by sutaakar** [comment]
@Kevin-Mok Can you please fix the build error? The health extension cannot be resolved in serverless-workflow-github-showcase as there is no bom imported.

**Comment by Kevin-Mok** [comment]
The check seems to be failing on the Spring Boot example for onboarding. I made [these changes](https://github.com/kiegroup/kogito-examples/pull/538/files#diff-23c1e48ecfdb7e05de0a44264b8e42bfabdc96202bd16874eca21c738b03d01d) to that particular POM, which were just from auto-generation. I don't think they're the reason for the failure though.

*Edit*: @sutaakar helped me figure out that the Spring Boot portion of the onboarding example is incompatible with the health extension, so I just removed it from the parent module and added it to all the other submodules except Spring Boot.

File: serverless-workflow-temperature-conversion/pom.xml [hunk 6]
```diff
@@ -1,20 +1,35 @@
...
     <module>subtraction-service</module>
     <module>multiplication-service</module>
     <module>conversion-workflow</module>
   </modules>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-universe-bom</artifactId>
+        <version>1.11.1.Final</version>
```
Comment (cristianonicolai) [inline serverless-workflow-temperature-conversion/pom.xml]: this dependency is probably not needed, each of the services already defines a dependency to quarkus, just add the smallrye-helath in there

File: serverless-workflow-github-showcase/pom.xml (hunk 1 above)
Comment (cristianonicolai) [reply serverless-workflow-github-showcase/pom.xml]: +1, this dependency is not needed, please add into the submodules directly.

**Review by cristianonicolai — CHANGES_REQUESTED** [review]
@Kevin-Mok it doesnt haelp much the review with so many changes not directly related to the PR goal. I would suggest next time send a PR just for reformat things separately.

File: serverless-workflow-github-showcase/pom.xml (hunk 1 above)
Comment (Kevin-Mok) [reply serverless-workflow-github-showcase/pom.xml]: @ricardozanini suggested adding it to [only the parent modules](https://github.com/kiegroup/kogito-examples/pull/538#discussion_r566160859), but the build fails if you remove this `quarkus-universe-bom` dependency as @sutaakar [pointed out](https://github.com/kiegroup/kogito-examples/pull/538#issuecomment-769778993).  

```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for io.quarkus:quarkus-smallrye-health:jar is missing. @ line 19, column 17
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.kie.kogito:serverless-workflow-temperature-conversion:2.0.0-SNAPSHOT (/home/kevin/coding/kogito/kogito-examples/serverless-workflow-temperature-conversion/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for io.quarkus:quarkus-smallrye-health:jar is missing. @ line 19, column 17
```

@ricardozanini, should I just move it into the submodules to get rid of this dependency then?

Comment (ricardozanini) [reply serverless-workflow-github-showcase/pom.xml]: Yeah, add them directly then.

**Comment by ricardozanini** [comment]
@cristianonicolai can you take a second look?

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**